erpnext/stock/utils.py

```diff
@@ -552,7 +554,16 @@ def check_pending_reposting(posting_date: str, throw_error: bool = True) -> bool
 
 
 @frappe.whitelist()
-def scan_barcode(search_value: str) -> Dict[str, Optional[str]]:
+def scan_barcode(search_value: str) -> BarcodeScanResult:
+	def set_cache(data: BarcodeScanResult):
+		frappe.cache().set(f"erpnext:barcode_scan:{search_value}", json.dumps(data), 120)
+
+	def get_cache() -> Optional[BarcodeScanResult]:
+		if data := frappe.cache().get(f"erpnext:barcode_scan:{search_value}"):
```

Comment:
`redis.set` / `redis.get` shouldn't be used (they don't respect multitenancy) use set_value / get_value instead.